### PR TITLE
Selection should not change when switching back from bView to pView #684

### LINKED ISSUE #684: Selection should not change when switching back from bView to pView
Currently, the selection seems to be based on index rather than a unique identifier. For example, if the 3rd issue was selected originally, the 3rd issue will be selected after switching to pView even if the originally selected issue is no longer the 3rd issue.

### PR DESCRIPTION
**Pull request by lejolly**
Fixes #684

### CODE REVIEW

**Comment by lejolly** [comment]
@damithc could you check the branch out and see if it fixes the problem

**Comment by damithc** [comment]
Is there anyway to write a test for this?

**Comment by lejolly** [comment]
We could try to expose the currently selected index, change the last updated time of an item, force the ui to refresh, and see if the index changes (it should assuming the current filter allows for it). 






On Wed, Jun 10, 2015 at 3:29 AM -0700, "Damith C. Rajapakse" <notifications@github.com> wrote:












Is there anyway to write a test for this?

**Comment by damithc** [comment]
Yes, let's try that. Preferably, write the test in a separate branch, make sure it fails, and then merge it to this branch and see if it passes.

**Comment by lejolly** [comment]
hmm, having some trouble trying to reproduce the bug in test mode

**Comment by damithc** [comment]
Can repro manually?

**Comment by damithc** [comment]
I used the filter `updated:5 sort:updated` to repro manually: select the top issue (there should be more than 1 issue in the panel), change to bView, do some change to the selected issue (e.g add comment/label), switch back to pView.

**Comment by lejolly** [comment]
yes i can reproduce it when using it with a real repo

**Comment by lejolly** [comment]
test now passes with the fix (and fails without the fix), works on travis too.

File: src/main/java/ui/components/NavigableListView.java [hunk 1]
```diff
@@ -113,6 +114,15 @@ public void restoreSelection() {
         }
     }
 
+    // sample generic findIndex, most of the time using the equals method
+    // will not give the desired behaviour because an item's properties might
+    // change but the item is still considered to be the same
+//    boolean areItemsEqual(T item1, T item2) {
+//        return item1.equals(item2);
+//    }
```
Comment (damithc) [inline src/main/java/ui/components/NavigableListView.java]: Clean up needed here.

**Pull request merged**